jablib/src/test/java/org/jabref/logic/quality/consistency/BibliographyConsistencyCheckTest.java

```diff
@@ -143,12 +179,87 @@ void unsetFieldsReported() {
                 .withCitationKey("withoutDate")
                 .withField(StandardField.URLDATE, "urldate");
 
+        List<BibEntry> bibEntriesList = List.of(withDate, withoutDate);
+        BibDatabase bibDatabase = new BibDatabase(bibEntriesList);
+        bibDatabase.insertEntries(bibEntriesList);
+
+        BibDatabaseContext bibContext = new BibDatabaseContext(bibDatabase);
+        bibContext.setMode(BibDatabaseMode.BIBLATEX);
+
         BibliographyConsistencyCheck.Result result = new BibliographyConsistencyCheck()
-                .check(List.of(withDate, withoutDate), (_, _) -> { });
+                .check(bibContext, (_, _) -> { });
 
         BibliographyConsistencyCheck.EntryTypeResult typeResult =
                 result.entryTypeToResultMap().get(StandardEntryType.Online);
 
         assertEquals(List.of(withDate, withoutDate), typeResult.sortedEntries().stream().toList());
     }
+
+    @Test
+    void unsetFieldsReportedInBibtexMode() {
+        BibEntry withDate = new BibEntry(StandardEntryType.Online)
+                .withCitationKey("withDate")
+                .withField(StandardField.DATE, "date")
+                .withField(StandardField.URLDATE, "urldate");
+        BibEntry withoutDate = new BibEntry(StandardEntryType.Online)
+                .withCitationKey("withoutDate")
+                .withField(StandardField.URLDATE, "urldate");
+
+        List<BibEntry> bibEntriesList = List.of(withDate, withoutDate);
+        BibDatabase bibDatabase = new BibDatabase(bibEntriesList);
+        bibDatabase.insertEntries(bibEntriesList);
+
+        BibDatabaseContext bibContext = new BibDatabaseContext(bibDatabase);
+        bibContext.setMode(BibDatabaseMode.BIBTEX);
+        BibliographyConsistencyCheck.Result result = new BibliographyConsistencyCheck()
+                .check(bibContext, (_, _) -> { });
+
+        assertEquals(Map.of(), result.entryTypeToResultMap());
+    }
+
+    @Test
+    void checkFieldEntriesWithFieldDifferences() {
+        Set<BibEntry> entries = new HashSet<>();
+        Set<Field> differingFields = Set.of(
+                StandardField.TITLE,
+                StandardField.PAGES,
+                new UnknownField("customField"),
+                StandardField.PUBLISHER
+        );
+
+        BibEntry entry1 = new BibEntry(StandardEntryType.Article, "id1")
+                .withField(StandardField.AUTHOR, "Autore Uno")
```

Comment:
I like more diversity in the code. This is inderstandable for most persons, thus, it is OK to leave it in.

In general, English language should be used also in examples -. if the exmaples do not touch some non-English language handling.